cmd/gitserver/server/server_grpc.go

```diff
@@ -101,3 +102,78 @@ func (gs *GRPCServer) Search(req *proto.SearchRequest, ss proto.GitserverService
 		},
 	})
 }
+
+func (gs *GRPCServer) Archive(req *proto.ArchiveRequest, ss proto.GitserverService_ArchiveServer) error {
+	//TODO(mucles): re-enable access logging (see server.go handleArchive)
+	// Log which which actor is accessing the repo.
+	// accesslog.Record(ctx, req.Repo,
+	// 	log.String("treeish", req.Treeish),
+	// 	log.String("format", req.Format),
+	// 	log.Strings("path", req.Pathspecs),
+	// )
+
+	if err := checkSpecArgSafety(req.Treeish); err != nil {
+		return status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	if req.Repo == "" || req.Format == "" {
+		return status.Error(codes.InvalidArgument, "empty repo or format")
+	}
+
+	execReq := &protocol.ExecRequest{
+		Repo: api.RepoName(req.Repo),
+		Args: []string{
+			"archive",
+			"--worktree-attributes",
+			"--format=" + req.Format,
+		},
+	}
+
+	if req.Format == string(gitserver.ArchiveFormatZip) {
```

Comment:
We should be using the generated
We should be using the generated getters when acessing _any_ fields from a protobuf struct. This is because all fields are optional in protobuf 3. 

See https://handbook.sourcegraph.com/departments/engineering/dev/tools/grpc_faq/#how-do-i-set-a-required-field-in-protobuf and https://protobuf.dev/reference/go/go-generated/#singular-scalar-proto3.

```suggestion
	if req.GetFormat() == string(gitserver.ArchiveFormatZip) {
```